src/Eccube/Controller/ShoppingController.php

```diff
@@ -1153,25 +1171,43 @@ public function shippingMultiple(Application $app, Request $request)
 
                         // 選択された情報を取得
                         $data = $item['customer_address']->getData();
+                        $CustomerAddress = null;
+                        $cusAddId = null;
                         if ($data instanceof CustomerAddress) {
-                            // 会員の場合、CustomerAddressオブジェクトを取得
-                            $CustomerAddress = $data;
+                            // Has been shipping check
+                            if (array_key_exists($cusAddId = $data->getId(), $arrCusAddTmp)) {
+                                // 会員の場合、CustomerAddressオブジェクトを取得
+                                $CustomerAddress = $data;
+                            }
                         } else {
-                            // 非会員の場合、選択されたindexが取得される
-                            $customerAddresses = $app['session']->get($this->sessionCustomerAddressKey);
-                            $customerAddresses = unserialize($customerAddresses);
-                            $CustomerAddress = $customerAddresses[$data];
-                            $pref = $app['eccube.repository.master.pref']->find($CustomerAddress->getPref()->getId());
-                            $CustomerAddress->setPref($pref);
+                            // Has been shipping check
+                            if (array_key_exists($cusAddId = $data, $arrCusAddTmp)) {
+                                // 非会員の場合、選択されたindexが取得される
+                                $customerAddresses = $app['session']->get($this->sessionCustomerAddressKey);
+                                $customerAddresses = unserialize($customerAddresses);
+                                $CustomerAddress = $customerAddresses[$data];
```

Comment:
`$customerAddresses[$data]` is what about unifind to `$customerAddresses[$cusAddId]`